src/analyzer/analyzermanager.cpp

```diff
@@ -0,0 +1,298 @@
+#include "analyzer/analyzermanager.h"
+
+#include <typeinfo>
+#include <QThread>
+
+#include <QtDebug>
+#include <QMutexLocker>
+#include <QListIterator>
+
+#include "library/trackcollection.h"
+#include "mixer/playerinfo.h"
+#include "track/track.h"
+#include "util/compatibility.h"
+#include "util/event.h"
+#include "util/timer.h"
+#include "util/trace.h"
+
+AnalyzerManager* AnalyzerManager::m_pAnalyzerManager = NULL;
+
+//static
+AnalyzerManager& AnalyzerManager::getInstance(UserSettingsPointer pConfig) {
+    if (!m_pAnalyzerManager) {
+        // There exists only one UserSettingsPointer in the app, so it doens't matter
+        // if we only assign it once.
```

Comment:
I would prefer this not become a singleton and instead pass it throughout to all the places it needs to go. Once https://github.com/mixxxdj/mixxx/pull/941 is in, there will be a better place for it in the Core class but for now it can go in the main window (src/mixxx.cpp).